Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -191,6 +191,83 @@ TEST_F(SerializerTest, EncodesStringModelToBytes) {
   }
 }
 
+TEST_F(SerializerTest, EncodesEmptyMapToBytes) {
+  FieldValue model = FieldValue::ObjectValue({});
+  // TEXT_FORMAT_PROTO: 'map_value: {}'
+  std::vector<uint8_t> bytes{0x32, 0x00};
+  ExpectRoundTrip(model, bytes, FieldValue::Type::Object);
+}
+
+TEST_F(SerializerTest, EncodesNestedObjectsToBytes) {
+  // As above, verify max int64_t value.
+  EXPECT_EQ(9223372036854775807, std::numeric_limits<int64_t>::max());
+
+  FieldValue model = FieldValue::ObjectValue(
+      {{"b", FieldValue::TrueValue()},
+       // TODO(rsgowman): add doubles (once they're supported)
+       // {"d", FieldValue::DoubleValue(std::numeric_limits<double>::max())},
+       {"i", FieldValue::IntegerValue(1)},
```

Comment:
I understand it might be preferable to be explicit here, but in general, does our `FieldValue` interface require calling the static "constructors"? I.e., is it not possible to have:
```
{"i", 1}, {"n", FieldValue::NullValue()}, {"s", "foo"}, ...
```